iowrappers/redis_client.go

```diff
@@ -268,46 +268,58 @@ func (redisClient *RedisClient) CacheLocationAlias(context context.Context, quer
 	return
 }
 
-// GetLocationWithAlias retrieves corrected location name from Redis; returns empty string if not exist;
-// corrects geocode query if corrected location name exists
-func (redisClient *RedisClient) GetLocationWithAlias(context context.Context, query *GeocodeQuery) string {
+func (redisClient *RedisClient) GetLocationWithAlias(context context.Context, query *GeocodeQuery) (string, error) {
+	Logger.Debugf("(RedisClient)GetLocationWithAlias -> request: %+v", *query)
 	var err error
-	resCity, err := redisClient.client.HGet(context, "location_name_alias_mapping:city_names", strings.ToLower(query.City)).Result()
-	if err != nil {
-		return ""
+	var resCity, resAdminAreaLevelOne, resCountry string
+	var locationSegments []string
+	if strings.TrimSpace(query.City) != "" {
```

Comment:
So my interpretation about the new IF clause can catch any scenario that query string is either empty, or has multiple spaces. It makes sense to me.
` if strings.TrimSpace()!="" { } `